components/datepicker/index.css

```
@@ -11,140 +11,186 @@ governing permissions and limitations under the License.
+  --spectrum-datepicker-input-width-quiet: calc(var(--spectrum-datepicker-range-input-width-quiet-first) + var(--spectrum-datepicker-icon-size) + var(--spectrum-datepicker-initial-height));
+
+  --spectrum-datepicker-range-input-width-first: calc(var(--spectrum-datepicker-initial-width) - 2 * var(--spectrum-datepicker-generic-padding));
+  --spectrum-datepicker-range-input-width-quiet-first: calc(var(--spectrum-datepicker-width-quiet-first) + var(--spectrum-datepicker-width-quiet-second));
+
+  --spectrum-datepicker-datetime-input-width-first: calc(var(--spectrum-datepicker-input-width-base) + var(--spectrum-datepicker-datetime-width-first));
+  --spectrum-datepicker-datetime-input-width: calc(var(--spectrum-datepicker-datetime-input-width-first) + var(--spectrum-datepicker-icon-size) + var(--spectrum-datepicker-initial-height));
+  --spectrum-datepicker-datetime-quiet-input-width-first: calc(var(--spectrum-datepicker-input-width-base) + var(--spectrum-datepicker-input-datetime-width));
+  --spectrum-datepicker-datetime-quiet-input-width: calc(var(--spectrum-datepicker-datetime-quiet-input-width-first) + var(--spectrum-datepicker-icon-size) + var(--spectrum-datepicker-initial-height));
+
+  /* padding inline end */
+  --spectrum-datepicker-padding-inline-end:  calc(var(--spectrum-datepicker-pickerbutton-width) + var(--spectrum-component-edge-to-text-100) - (var(--spectrum-datepicker-border-width) * 2));
+  --spectrum-datepicker-padding-inline-end-quiet: calc((var(--spectrum-datepicker-pickerbutton-width) + var(--spectrum-component-edge-to-text-100)) - var(--spectrum-datepicker-quiet-button-offset));
+  --spectrum-datepicker-padding-inline-end-invalid: calc(var(--spectrum-datepicker-pickerbutton-width) + var(--spectrum-component-edge-to-text-100) + var(--spectrum-datepicker-icon-to-button) + var(--spectrum-datepicker-icon-size) - (var(--spectrum-datepicker-border-width) * 2));
+  --spectrum-datepicker-padding-inline-end-invalid-quiet: calc(var(--spectrum-datepicker-pickerbutton-width-quiet) + var(--spectrum-datepicker-icon-size) + var(--spectrum-datepicker-icon-to-text));
 }
 
 .spectrum-DatePicker {
   position: relative;
   display: inline-flex;
   flex-direction: row;
   flex-wrap: nowrap;
-  min-inline-size: var(--spectrum-alias-single-line-width);
-  border-radius: var(--spectrum-alias-border-radius-regular);
+  min-inline-size: var(--mod-datepicker-min-width, var(--spectrum-datepicker-min-width));
+  border-radius: var(--mod-datepicker-border-radius, var(--spectrum-datepicker-border-radius));
+
+  &.is-disabled {
+    .spectrum-DatePicker-rangeDash {
+      color: var(--highcontrast-datepicker-dash-color-disabled, var(--mod-datepicker-dash-color-disabled, var(--spectrum-datepicker-dash-color-disabled)))
+    }
+  }
+  &.is-invalid {
+    /* positions invalid icon to not be covered by the picker button */
+    .spectrum-Textfield .spectrum-Textfield-validationIcon {
```

```suggestion
    .spectrum-Textfield .spectrum-Textfield-validationIcon {
```

This shouldn't be necessary - you can set the mod values at .is-invalid and it should cascade no problem - esp since we're hooking into the mods. I think ideally we also try to avoid using classes for other components in this one since it won't convert well when parsed for web components.